src/Dictionaries/IDictionary.cpp

```diff
@@ -0,0 +1,154 @@
+#include <Dictionaries/IDictionary.h>
+
+namespace DB
+{
+
+void IDictionary::convertKeyColumns(Columns & key_columns, DataTypes & key_types) const
```

Comment:
When I tried to move some IDictionary methods into cpp there were some linking error, because module with Storage doesn't depend on Dictionaries, if I understood it correctly.

Reports from previous PRs:
https://s3.amazonaws.com/clickhouse-builds/38956/35d7acfb7b0b15a8a87ee50ada2cdb4b7cbdedfe/clickhouse_special_build_check/report.html
https://s3.amazonaws.com/clickhouse-builds/38956/35d7acfb7b0b15a8a87ee50ada2cdb4b7cbdedfe/clickhouse_build_check/report.html